tests/test_patches.py

```diff
@@ -110,15 +217,56 @@
 
     # session.py
     Session.__init__: {'ccf156a76beda3425fb54363f3b2718dc0445f6d'},
+    Session._register_response_parser_factory:
+        {'d6cd5a8b1b473b0ec3b71db5f621acfb12cc412c'},
     Session.create_client: {'36f4e718fc4bada66808c2f98fa71835c09076f7'},
+    Session._create_credential_resolver: {'87e98d201c72d06f7fbdb4ebee2dce1c09de0fb2'},
+    Session.get_credentials: {'c0de970743b6b9dd91b5a71031db8a495fde53e4'},
     get_session: {'c47d588f5da9b8bde81ccc26eaef3aee19ddd901'},
 
+    # signers.py
+    RequestSigner.handler: {'371909df136a0964ef7469a63d25149176c2b442'},
+    RequestSigner.sign: {'7df841d3df3f4015763523c1932652aef754287a'},
+    RequestSigner.get_auth: {'4f8099bef30f9a72fa3bcaa1bd3d22c4fbd224a8'},
+    RequestSigner.get_auth_instance: {'4f8099bef30f9a72fa3bcaa1bd3d22c4fbd224a8'},
+    RequestSigner._choose_signer: {'d1e0e3196ada449d3ae0ec09b8ae9b5868c50d4e'},
+    RequestSigner.generate_presigned_url: {'2acffdfd926b7b6f6cc4b70b90c0587e7f424888'},
+    add_generate_presigned_url: {'5820f74ac46b004eb79e00eea1adc467bcf4defe'},
+    generate_presigned_url: {'9c471f957210c0a71a11f5c73be9fed844ecb5bb'},
+
+    # utils.py
+    ContainerMetadataFetcher.__init__:
+        {'46d90a7249ba8389feb487779b0a02e6faa98e57'},
+    ContainerMetadataFetcher.retrieve_full_uri:
+        {'2c7080f7d6ee5a3dacc1b690945c045dba1b1d21'},
+    ContainerMetadataFetcher.retrieve_uri:
+        {'4ee8aa704cf0a378d68ef9a7b375a1aa8840b000'},
+    ContainerMetadataFetcher._retrieve_credentials:
+        {'f5294f9f811cb3cc370e4824ca106269ea1f44f9'},
+    ContainerMetadataFetcher._get_response:
+        {'7e5acdd2cf0167a047e3d5ee1439565a2f79f6a6'},
+    # Overrided session and dealing with proxy support
+    IMDSFetcher.__init__: {'690e37140ccdcd67c7a85ce5d36331491a79954e'},
+    IMDSFetcher._get_request: {'96a0e580cab5a21deb4d2cd7e904aa17d5e1e504'},
+    IMDSFetcher._fetch_metadata_token: {'4fdad673b4997b1268c6d9dff09a4b99c1cb5e0d'},
+
+    InstanceMetadataFetcher.retrieve_iam_role_credentials:
+        {'76737f6add82a1b9a0dc590cf10bfac0c7026a2e'},
+    InstanceMetadataFetcher._get_iam_role: {'80073d7adc9fb604bc6235af87241f5efc296ad7'},
+    InstanceMetadataFetcher._get_credentials:
+        {'1a64f59a3ca70b83700bd14deeac25af14100d58'},
+
     # waiter.py
     NormalizedOperationMethod.__call__: {'79723632d023739aa19c8a899bc2b814b8ab12ff'},
     Waiter.wait: {'5502a89ed740fb5d6238a6f72a3a08efc1a9f43b'},
     create_waiter_with_client: {'c3d12c9a4293105cc8c2ecfc7e69a2152ad564de'},
 }
 
+# REPLACE = backwards incompatible change
+# APPEND = officially supporting more versions of botocore/aiohttp
+
+# If you're changing these, most likely need to update setup.py as well.
+
 _PROTOCOL_PARSER_CONTENT = {'ec2', 'query', 'json', 'rest-json', 'rest-xml'}
 
 
```

Comment:
@terrycain wow =)

Reply:
Yeah... the rabbit hole ran deep on this one. Mainly the credentials class and the utils that it used.